### ISSUE TITLE
"UnityActionException: The episode is completed." with --num-envs > 1

### ISSUE THREAD

**Issue by WHiT3R4bBiT** [Unity-Technologies/ml-agents#1961]
Running v0.8.1 I get the following error when `--num-envs` is greater than 1. The error happens at the step where AcademyReset() should be called (30k steps in our case).

```Traceback (most recent call last):
  File "/usr/lib64/python3.6/multiprocessing/process.py", line 249, in _bootstrap
    self.run()
  File "/usr/lib64/python3.6/multiprocessing/process.py", line 93, in run
    self._target(*self._args, **self._kwargs)
  File "/home/ec2-user/ml-agents-envs/mlagents/envs/subprocess_environment.py", line 64, in worker
    all_brain_info = env.step(vector_action, memory, text_action, value)
  File "/home/ec2-user/ml-agents-envs/mlagents/envs/environment.py", line 447, in step
    "The episode is completed. Reset the environment with 'reset()'")
mlagents.envs.exception.UnityActionException: The episode is completed. Reset the environment with 'reset()'```

**Comment by vincentpierre** [comment]
Hi @WHiT3R4bBiT 
I was able to reproduce your bug. I modified the *3DBall* environment Academy script this way : 

```csharp
using System.Collections;
using System.Collections.Generic;
using UnityEngine;
using MLAgents;

public class Ball3DAcademy : Academy
{
    int numToReset = 10;
    int numStep = 0;
    public override void AcademyReset()
    {
    }

    public override void AcademyStep()
    {
        numStep ++;
        if ( numStep > numToReset){
            Done();
            numToReset = Random.Range(100,200);
            numStep = 0;
        }
    }
}
```

This error seems to arise when the the environments do not need to reset at the same time. One of them needs to reset but not the other. In this situation, a STEP command might be sent while one of the environment was expecting a reset leading to the error you saw. 
This is a bug, thank you for posting this issue. We will look into it. In the meantime, I would recommend not using `Academy.Done` at all in your environment and rely on `Academy max step` or `Agent.Done`.

**Comment by harperj** [comment]
This issue was fixed in #2067.  If you use our `develop` branch you can see the fix today, or it will be in the next ML-Agents release.  Sorry for the inconvenience.  I'm going to close this issue now, but please feel free to reopen or create a new issue if you have any more trouble.
